### PR TITLE
Logging configuration help

### PR DESCRIPTION
**Pull request by paolocappelletti**

### CODE REVIEW

**Comment by paolocappelletti** [comment]
see also this related PR: https://github.com/HorizenOfficial/Sidechains-SDK/pull/418

File: docs/introduction/Sidechain-SDK-extension.rst [hunk 1]
```diff
@@ -609,4 +609,59 @@ The Backup Storage must be present inside your node data directory before starts
...
+        Logger logger = LogManager.getLogger(MyCustomClass.class);
+
+        public MyCustomClass(){
+          logger.debug("This is an example  debug message inside the constructor");
+        }
+    }
+
+Note: do not add any log4j library in your application pom file, as it is already loaded as a nested dependency of the SDK. 
+
+You can rely on the default logging configuration (based on Log4J library) and change just a few parameters inside the application config file, or ovveride it completely with a custom one.
```
Comment (MarcoOl94) [inline docs/introduction/Sidechain-SDK-extension.rst:633]: config -> configuration

File: docs/introduction/Sidechain-SDK-extension.rst [hunk 2]
```diff
@@ -609,4 +609,59 @@ The Backup Storage must be present inside your node data directory before starts
...
+
+Default log configuration
+-------------------------------
+By default, `a predefined log4j2.xml configuration <https://github.com/HorizenOfficial/Sidechains-SDK/blob/master/sdk/src/main/resources/log4j2.xml>`_  is used.
+It redirects all logging messages to the system console and to a filesystem log file, rotated and gzipped when it reaches 50MB size (only the latest 10 are then retained).
+
+The following dynamic parameters coming from the application conf file are used, so that you can change them by just updating the application conf file:
```
Comment (MarcoOl94) [inline docs/introduction/Sidechain-SDK-extension.rst:640]: We repeat twice "application conf", maybe we can rephrase in something like "The following dynamic parameters are taken from the application configuration file but you can change them at any time."

File: docs/introduction/Sidechain-SDK-extension.rst [hunk 3]
```diff
@@ -609,4 +609,59 @@ The Backup Storage must be present inside your node data directory before starts
...
+
+        logInfo {
+          logFileName = "debug.log"
+          logFileLevel = "warn"
+          logConsoleLevel = "debug"
+        }
+
+        ...
+
+- ``scorex.logDir``: base folder where log files are generated, injected in the log4jxml in the placeholder ``${sys:logDir}``
```
Comment (MarcoOl94) [inline docs/introduction/Sidechain-SDK-extension.rst:656]: "where log files" -> "where the log files"

**Review by MarcoOl94 — CHANGES_REQUESTED** [review]

File: docs/introduction/Sidechain-SDK-extension.rst (hunk 1 above)
Comment (i-Alex) [inline docs/introduction/Sidechain-SDK-extension.rst:633]: ovveride -> override

**Review by i-Alex — CHANGES_REQUESTED** [review]

**Review by i-Alex — APPROVED** [review]

**Review by MarcoOl94 — APPROVED** [review]

**Pull request merged**